src/components/navigation/Precedent.tsx

```diff
@@ -5,16 +5,44 @@ import { usePrevious } from '../../lib/commons/usePrevious';
 import { fr } from '@codegouvfr/react-dsfr';
 
 export function Precedent(props: OrchestratedElement) {
-	const { goPreviousPage = () => null, isFirstPage, pageTag } = props;
+	const {
+		goPreviousPage = () => null,
+		isFirstPage,
+		pageTag,
+		getComponents = () => [],
+	} = props;
 
 	const buttonRef = useRef<HTMLButtonElement>(null);
 	const previousPage = usePrevious<string>(pageTag);
 
 	useEffect(() => {
 		if (pageTag !== previousPage) {
-			buttonRef.current?.focus();
+			const components = getComponents();
+			const hasConfirmationModal = components.some(
+				(component) => component.componentType === 'ConfirmationModal'
+			);
+			if (hasConfirmationModal) {
+				const cancelButtons = Array.from(
+					document.querySelectorAll('button[value="cancel"]')
+				);
+
+				const handleCancelClick = () => {
+					setTimeout(() => {
+						const continueButton = document.getElementById('continue-button');
+						continueButton?.focus();
+						cancelButtons.forEach((cancelButton) => {
+							cancelButton.removeEventListener('click', handleCancelClick);
+						});
+					}, 0);
```

Comment:
un setTimeout avec  un délais de 0, ça revient à appeler la fonction directement ?
si oui, tu peux coller touit le contenu des lignes 31 à 35 à la racine.
c'est important de retirer l'eventListener ?

Reply:
Non c'est pas pareil : https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful

Oui c'est important car on a plusieurs boutons